bin/configure-tower-projects.py

```diff
@@ -528,14 +528,14 @@ def create_credentials(self) -> int:
                 assert cred["deleted"] is None
                 return cred["id"]
         # Otherwise, create a new credentials entry for the project
-        secret_arn = self.stack["TowerRoleArn"]
-        credentials = self.org.aws.get_secret_value(secret_arn)
+        # secret_arn = self.stack["TowerRoleArn"]
+        # credentials = self.org.aws.get_secret_value(secret_arn)
         data = {
             "credentials": {
                 "name": self.stack_name,
                 "provider": "aws",
                 "keys": {
-                    "accessKey": credentials["aws_access_key_id"],
+                    # "accessKey": credentials["aws_access_key_id"],
```

Comment:
i believe the key to setting this `key` is to look at line #544, `response = self.tower.request("POST", endpoint, params=params, json=data)`.  That is using the [seqera tower credentials API](https://cloud.seqera.io/openapi/index.html?_gl=1*yd5xxw*_gcl_au*MTIxMzIyMTA3My4xNzMwNDkyODI0#post-/credentials) to `Creates new Tower credentials`.  According to the API docs, passing in an `assumeRoleArn` also requires `accessKey` and `secretKey` therefore this will not work.

Reply:
Yes, this is the confusing part I have. I remembered it worked when keeping the `accessKey` and `secretKey`. However, as my understanding based on this [doc](https://docs.seqera.io/platform/23.4/enterprise/advanced-topics/use-iam-role#:~:text=create%20a%20new%20AWS%20credential.%20You%20are%20now%20prompted%20for%20an%20AWS%20arn%20instead%20of%20access%20keys.), the `accessKey` and `secretKey` should be no longer needed if implementing role based authentication. Instead, the AWS role arn is used after the IAM artifacts is created and new credentials are added in Seqera.

Reply:
Unfortunately this deployment setup does not fit with the assume role use case that's explained in the seqera docs.  This script `configure-tower-projects.py` is [executed with a Github action](https://github.com/Sage-Bionetworks-Workflows/nextflow-infra/blob/dev/.github/workflows/rw-deploy.yaml#L76-L80)  which means it is running in a Github supplied instance, NOT an AWS instance.  Only AWS instances are setup to assume roles that give access to the Seqera app (and APIs).